lib/blockchain.go

```diff
@@ -1907,20 +1912,21 @@ func (bc *Blockchain) ProcessBlock(desoBlock *MsgDeSoBlock, verifySignatures boo
 		return false, false, nil, fmt.Errorf("ProcessBlock: Block is nil")
 	}
 
-	// If the block's height is after the PoS cut-over fork height, then we use the PoS block processing logic. Otherwise, fall back
-	// to the PoW logic.
-	if desoBlock.Header.Height >= uint64(bc.params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight) {
-		// TODO: call bc.processBlockPoS(desoBlock, verifySignatures) instead
-		isMainChain, isOrphan, err := bc.processBlockPoW(desoBlock, verifySignatures)
-		return isMainChain, isOrphan, nil, err
+	// If the block's height is after the PoS cut-over fork height, then we use the PoS block processing logic.
+	// Otherwise, fall back to the PoW logic.
+	if bc.IsPoSBlockHeight(desoBlock.Header.Height) {
+		return bc.processBlockPoS(desoBlock, 1, verifySignatures)
```

Comment:
Did some more digging. It looks like you define a MsgDeSoHeader.GetView() that patches the view with the height for HeaderVersion1/. BUT the code in getLineageFromTip() still looks wrong because it's calling block.Header.ProposedInview. Will keep reading...

Reply:
Mystery solved. Before the line that checks ProposedInView in getCommittedTip(), there is this line, which breaks out of the for loop:
```
if currentBlock.Hash.IsEqual(highestCommittedBlock.Hash) {
    break
```

This results in an empty ancestors list being returned, which is what we expect.